File: src/main/java/br/com/kleberson2santos/casadocodigov1/novoautor/AutorRepository.java
package br.com.kleberson2santos.casadocodigov1.novoautor;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface AutorRepository extends CrudRepository<Autor, Long> {
    Optional<Autor> findByEmail(String email);
}

File: src/main/kotlin/com/tiozao/cdd/loja/repository/AutorRepository.kt
package com.tiozao.cdd.loja.repository

import com.tiozao.cdd.loja.repository.entity.AutorEntity
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface AutorRepository: CrudRepository<AutorEntity,Int>{

    fun findByEmail( email: String ): Optional<AutorEntity>
}